services/src/main/java/org/keycloak/social/apple/AppleIdentityProviderConfig.java

```diff
@@ -0,0 +1,47 @@
+package org.keycloak.social.apple;
+
+import org.keycloak.broker.oidc.OIDCIdentityProviderConfig;
+import org.keycloak.models.IdentityProviderModel;
+
+/**
+ * @author Emilien Bondu
+ */
+public class AppleIdentityProviderConfig extends OIDCIdentityProviderConfig {
```

Comment:
To automatically renew client secret in time, we tried something like this, and it seems to be working:

```java

    @Override
    public String getClientSecret() {
        String clientSecret = super.getClientSecret();
        if (isValidSecret(clientSecret)) {
            return clientSecret;
        } else {
            String nextSecret = generateJWS(
                    parseP8(this.getP8Content()),
                    this.getKeyId(),
                    this.getTeamId(),
                    this.getClientId(),
                    Duration.ofHours(24)
            );
            this.setClientSecret(nextSecret);
            return nextSecret;
        }
    }
```

This is not very keycloak-esque, is there anyway to update the providerconfig programmatically with the new client_secret ?

Though if `AppleIdentityProvider` is new-ed up on every login the current solution should be safe as well.

Reply:
We looked to renew the secret when expired but first, we need to fin a way to save the renewed key in the bd. Actually the secret is regenerated at each login.